src/Compilers/Core/Portable/DocumentationCommentId.cs

```diff
@@ -1374,7 +1374,12 @@ private static ITypeParameterSymbol GetNthTypeParameter(INamedTypeSymbol typeSym
                 }
 
                 var index = n - containingTypeParameterCount;
-                return typeSymbol.TypeParameters[index];
+                if (index < typeSymbol.TypeParameters.Length)
```

Comment:
>typeSymbol.TypeParameters [](http://example.com/codeflow?start=28&length=25)

Consider if caching this array is appropriate. It is quite possible that the property returns different arrays with the same content every time. The code is safe, but could allocate two arrays instead of one.

Reply:
Thanks, cached: 0900d3c79a4cdb651c5093f1ed91bccca4019c22